filebeat/_meta/common.reference.p2.yml

```diff
@@ -238,6 +238,16 @@ filebeat.prospectors:
   # Maximum size of the message received over UDP
   #max_message_size: 10240
 
+#------------------------------ Tcp prospector --------------------------------
+# Experimental: Config options for the TCP input
+#- type: tcp
+  #enabled: false
+  # The host and port to receive the new event
+  #host: "localhost:8080"
+
+  # Character used to split new message
+  #line_delimiter: "\n"
```

Comment:
Do we need this to be configurable? I would expect that same behaviour that supports also `\r` etc. like the log prospector / harvester.

Reply:
I believe it needs to be configurable; this input is more than just syslog entries.
If you look at the code by default, it will use the [ScanLines](https://golang.org/src/bufio/scan.go) function to deal with that case. Anything else will use our own implementation of the split.

The default method will detect `\n,` but it will also strip `\r`, I think this is the most common use case with plain sockets protocol. So I think we should be fine(tm)?

The custom delimiter is OK for more esoteric implementation. Maybe in a future revision, we could allow multiple patterns.

Reply:
The feature got requested already several times for the log harvester ...

Reply:
I haven't look at the log harvester how it does the splitting... maybe we can reuse it.

Reply:
I did a quick look at the log harvester, I _believe_ we could use the same scanner/split.